kong/clustering/version_negotiation/init.lua

```diff
@@ -354,4 +354,40 @@ function _M.get_negotiated_service(name)
   return val.version, val.message
 end
 
+--- detect '/v1/wrpc' endpoint
+--- if there is no '/v1/wrpc', fallback to websocket + json
+function _M.head_request_version(conf, cert, cert_key)
+  local params = {
+    scheme = "https",
+    method = "HEAD",
+
+    ssl_verify = false,
+    ssl_client_cert = cert,
+    ssl_client_priv_key = cert_key,
+  }
+
+  if conf.cluster_mtls == "shared" then
+    params.ssl_server_name = "kong_clustering"
+
+  else
+    -- server_name will be set to the host if it is not explicitly defined here
+    if conf.cluster_server_name ~= "" then
+      params.ssl_server_name = conf.cluster_server_name
+    end
+  end
+
+  local c = http.new()
+  local res, err = c:request_uri(
+    "https://" .. conf.cluster_control_plane .. "/v1/wrpc", params)
+  if not res then
+    return nil, err
+  end
+
+  if res.status == 404 then
+    return "v0", "no wrpc endpoint, fallback to /v1/outlet"
```

Comment:
I don't think this error message is even used and should just be removed. Plus why can't the function just be called "check_wrpc_support", `v0` and `v1` seems to be confusing and unnecessary.

Reply:
Yes. v0 and v1 mean nothing to a user. Maybe we could log like this: